test/unit/lib/plugins/aws/package/compile/events/schedule.test.js

```diff
@@ -1,295 +1,323 @@
-          .Resources.FirstEventsRuleSchedule1.Type
-      ).to.equal('AWS::Events::Rule');
-      expect(
-        awsCompileScheduledEvents.serverless.service.provider.compiledCloudFormationTemplate
-          .Resources.FirstEventsRuleSchedule2.Type
-      ).to.equal('AWS::Events::Rule');
-      expect(
-        awsCompileScheduledEvents.serverless.service.provider.compiledCloudFormationTemplate
-          .Resources.FirstEventsRuleSchedule3.Type
-      ).to.equal('AWS::Events::Rule');
-      expect(
-        awsCompileScheduledEvents.serverless.service.provider.compiledCloudFormationTemplate
-          .Resources.FirstLambdaPermissionEventsRuleSchedule1.Type
-      ).to.equal('AWS::Lambda::Permission');
-      expect(
-        awsCompileScheduledEvents.serverless.service.provider.compiledCloudFormationTemplate
-          .Resources.FirstLambdaPermissionEventsRuleSchedule2.Type
-      ).to.equal('AWS::Lambda::Permission');
-      expect(
-        awsCompileScheduledEvents.serverless.service.provider.compiledCloudFormationTemplate
-          .Resources.FirstLambdaPermissionEventsRuleSchedule3.Type
-      ).to.equal('AWS::Lambda::Permission');
-      expect(
-        awsCompileScheduledEvents.serverless.service.provider.compiledCloudFormationTemplate
-          .Resources.FirstLambdaPermissionEventsRuleSchedule4.Type
-      ).to.equal('AWS::Lambda::Permission');
-    });
+      },
+    },
+  };
+  const { awsNaming, cfTemplate } = await runServerless(params);
+  const cfResources = cfTemplate.Resources;
 
-    it('should respect enabled variable, defaulting to true', () => {
-      awsCompileScheduledEvents.serverless.service.functions = {
-        first: {
-          events: [
-            {
-              schedule: {
-                rate: 'rate(10 minutes)',
-                enabled: false,
-              },
-            },
-            {
-              schedule: {
-                rate: 'rate(10 minutes)',
-                enabled: true,
-              },
-            },
-            {
-              schedule: {
-                rate: 'rate(10 minutes)',
-              },
-            },
-            {
-              schedule: 'rate(10 minutes)',
-            },
-          ],
+  const scheduleCfResources = [];
+
+  for (const event of events) {
+    const schedule = event.schedule;
+    let scheduleEvents;
+
+    if (typeof schedule === 'string') {
+      scheduleEvents = [schedule];
+    } else {
+      scheduleEvents = Array.isArray(schedule.rate) ? schedule.rate : [schedule.rate];
+    }
+
+    for (let i = 0; i < scheduleEvents.length; i++) {
+      const index = scheduleCfResources.length + 1;
+
+      const scheduleLogicalId = awsNaming.getScheduleLogicalId('test', index);
+      const scheduleCfResource = cfResources[scheduleLogicalId];
+
+      scheduleCfResources.push(scheduleCfResource);
+    }
+  }
+
+  return scheduleCfResources;
+}
+
+describe('ScheduleEvents', () => {
```

Comment:
Wow, you've refactored all of the tests, not only the new ones, well done :clap: 

Let's instead of `ScheduleEvents` name this describe with the path to this test file as it's done in other places (it's not documented, but it's a convention that we've adopted a while ago).

Reply:
:heavy_check_mark: